### PR TITLE
Datamap Report: Rename columns

### PR DESCRIPTION
**Pull request by gilluminate**
Closes [HJ-1](https://ethyca.atlassian.net/browse/HJ-1)

### Description Of Changes

Fides Plus users should be able to edit the name of the columns in reporting views. User's changes will be saved locally in the browser, but will also be saved as part of custom reports. 

### Code Changes

* Edit Columns button moved as part of the new kebab menu (no other changes to this functionality)
* New Rename columns option under the kebab menu
* New Table v2 header type: `EditableHeader` to support edit mode here
* New column naming mappings to keep track of defaults/saved names
* refactor Datamap table to support new column naming as a form using Formik
* refactor Edit columns modal and Filters modal to support custom column names (as applicable)
* Update E2E tests to ensure quality of various features

### Steps to Confirm

1. Log in to Admin UI > Data map report (eg. http://localhost:3000/reporting/datamap)
2. Click the new kebab menu button (sideways ellipsis button)
3. In the popup menu click "Rename columns"
4. All column headers in the table should now be editable:
  a. Each input should have a placeholder which is the equivalent of the original column header name
  b. 3 new buttons should be present: "Reset All", "Cancel", and "Apply" (primary)
5. change the name of one or more columns by typing a new name in that header's input
6. apply your changes by clicking the "Apply" button
7. The headers should no longer be editable:
  a. Newly entered names should appear as the Header name of their respective columns
  b. 3 buttons are no longer visible
8. Click the menu and then the menu option to "Rename columns" again
9. Make some new changes to the column headers
10. Click the "Cancel" button
11. New changes should not be applied and the table is no longer in an editable state
12. Create a new Report with the new names applied to it
13. Click the menu and then the menu option to "Rename columns" again
14. This time click the "Reset all" button
15. All headers should be reverted to their default name and the table should not longer be editable
16. Apply the newly created Report
17. Custom names should be applied to the table again
18. Make sure one of the following headers has been customized: "Data use", "Data categories", or "Data subject" (or all 3 if you want)
19. Click the "Filter" button and make sure the customization applied in the previous step is reflected here
20. Close the Filter modal
21. Click the kebab and then "Edit columns" button
22. Make sure the customized header names are reflected here

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* [x] Issue Requirements are Met
* [x] Update `CHANGELOG.md`

[HJ-1]: https://ethyca.atlassian.net/browse/HJ-1?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: clients/admin-ui/src/features/common/custom-reports/CustomReportTemplates.tsx [hunk 1]
```diff
@@ -223,10 +220,6 @@ export const CustomReportTemplates = ({
   const applyDisabled =
     (!fetchedReport && !savedReportId) || fetchedReport?.id === savedReportId;
 
-  if (!userCanSeeReports) {
-    return null;
-  }
-
```
Comment (gilluminate) [inline clients/admin-ui/src/features/common/custom-reports/CustomReportTemplates.tsx:229]: I moved this logic to the parent component so none of this loads if the user doesn't have permissions to view it.

File: clients/admin-ui/src/features/common/table/v2/filters/GlobalFilterV2.tsx [hunk 2]
```diff
@@ -35,7 +35,7 @@ export const GlobalFilterV2 = ({
   }, [value, onClear]);
 
   return (
-    <Box maxWidth="510px" width="100%">
+    <Box maxWidth="424px" width="100%">
```
Comment (gilluminate) [inline clients/admin-ui/src/features/common/table/v2/filters/GlobalFilterV2.tsx:38]: matches newer designs and leaves more room for the extra buttons

File: clients/admin-ui/src/features/datamap/constants.ts [hunk 3]
```diff
@@ -61,16 +61,3 @@ COLUMN_NAME_MAP[
...
-  COLUMN_ORDER = "datamap-column-order",
-  COLUMN_VISIBILITY = "datamap-column-visibility",
-  COLUMN_SIZING = "datamap-column-sizing",
-  COLUMN_EXPANSION_STATE = "datamap-column-expansion-state",
-  CUSTOM_REPORT_ID = "datamap-custom-report-id",
-  FILTERS = "datamap-filters",
-  GROUP_BY = "datamap-group-by",
-  SORTING_STATE = "datamap-sorting-state",
-  WRAPPING_COLUMNS = "datamap-wrapping-columns",
-}
```
Comment (gilluminate) [inline clients/admin-ui/src/features/datamap/constants.ts:76]: moved to new reporting constants file

File: clients/admin-ui/src/features/datamap/datamap-drawer/SystemInfo.tsx [hunk 4]
```diff
@@ -114,7 +114,7 @@ const SystemInfo = ({ system }: SystemInfoProps) => {
               </Box>
               <Box marginTop={3}>
                 <CustomTextArea
-                  label="System Description"
+                  label="System description"
```
Comment (gilluminate) [inline clients/admin-ui/src/features/datamap/datamap-drawer/SystemInfo.tsx:117]: just a drive-by fix

File: clients/admin-ui/src/features/datamap/reporting/DatamapReportTableColumns.tsx [hunk 5]
```diff
@@ -1,74 +1,50 @@
...
-  SYSTEM_DEPENDENCIES = "system_dependencies",
-  THIRD_COUNTRY_SAFEGUARDS = "third_country_safeguards",
-  THIRD_PARTIES = "third_parties",
-  COOKIES = "cookies",
-  USES_COOKIES = "uses_cookies",
-  USES_NON_COOKIE_ACCESS = "uses_non_cookie_access",
-  USES_PROFILING = "uses_profiling",
-  SYSTEM_UNDECLARED_DATA_CATEGORIES = "system_undeclared_data_categories",
-  DATA_USE_UNDECLARED_DATA_CATEGORIES = "data_use_undeclared_data_categories",
-}
```
Comment (gilluminate) [inline clients/admin-ui/src/features/datamap/reporting/DatamapReportTableColumns.tsx:66]: Moved to new reporting constants file

File: clients/admin-ui/src/features/datamap/reporting/DatamapReportTableColumns.tsx [hunk 6]
```diff
@@ -93,10 +69,8 @@ const getCustomFieldColumns = (
     // We need to figure out the original custom field object in order to see
     // if the value is a string[], which would want `showHeaderMenu=true`
     const customField = customFields.find((cf) =>
-      key.includes(_.snakeCase(cf.name)),
+      key.includes(snakeCase(cf.name)),
     );
-    const keyWithoutPrefix = key.replace(/^(system_|privacy_declaration_)/, "");
-    const displayText = _.upperFirst(keyWithoutPrefix.replaceAll("_", " "));
```
Comment (gilluminate) [inline clients/admin-ui/src/features/datamap/reporting/DatamapReportTableColumns.tsx:99]: moved to a util

File: clients/admin-ui/src/features/privacy-experience/PrivacyExperiencesTable.tsx [hunk 7]
```diff
@@ -150,7 +150,6 @@ export const PrivacyExperiencesTable = () => {
           ),
           header: (props) => <DefaultHeaderCell value="Locations" {...props} />,
           meta: {
-            displayText: "Locations",
```
Comment (gilluminate) [inline clients/admin-ui/src/features/privacy-experience/PrivacyExperiencesTable.tsx:153]: a little "Cargo Cult Coding" happening here, these `displayText` entries were never used on any table except for the Datamap report

**Review by jpople — APPROVED** [review]
✅ Tested locally, everything working as expected for me.  Great work!
